src/charm.py

```diff
@@ -28,142 +40,91 @@ def __init__(self, msg: str, status_type=None):
-                            "USERID_HEADER": "kubeflow-userid",
-                            "USERID_PREFIX": "",
-                            "PROFILES_KFAM_SERVICE_HOST": f"{profiles_service}.{model}",
-                            "REGISTRATION_FLOW": config["registration-flow"],
-                            "DASHBOARD_LINKS_CONFIGMAP": config["dashboard-configmap"],
-                        },
-                        "ports": [{"name": "ui", "containerPort": config["port"]}],
-                        "kubernetes": {
-                            "livenessProbe": {
-                                "httpGet": {"path": "/healthz", "port": config["port"]},
-                                "initialDelaySeconds": 30,
-                                "periodSeconds": 30,
-                            }
-                        },
-                    }
-                ],
-            },
-            {
-                "configMaps": {
-                    config["dashboard-configmap"]: {
-                        "settings": json.dumps(
-                            {
-                                "DASHBOARD_FORCE_IFRAME": True,
-                            }
-                        ),
-                        "links": Path("src/config.json").read_text(),
-                    },
-                },
-                "kubernetesResources": {
-                    "customResources": {
-                        "profiles.kubeflow.org": [
-                            {
-                                "apiVersion": "kubeflow.org/v1beta1",
-                                "kind": "Profile",
-                                "metadata": {"name": config["profile"]},
-                                "spec": {
-                                    "owner": {"kind": "User", "name": config["profile"]}
-                                },
-                            }
-                        ]
+    def get_kubeflow_dashboard_operator_layer(self, profiles_service: str) -> Layer:
+        """Returns a pre-configured Pebble layer."""
+        layer_config = {
+            "summary": "dex-auth-operator layer",
+            "description": "pebble config layer for kubeflow_dashboard_operator",
+            "services": {
+                self._container_name: {
+                    "override": "replace",
+                    "summary": "entrypoint of the kubeflow_dashboard_operator image",
+                    "command": f"{self._entrypoint}",
+                    "startup": "enabled",
+                    "environment": {
+                        "USERID_HEADER": "kubeflow-userid",
+                        "USERID_PREFIX": "",
+                        "PROFILES_KFAM_SERVICE_HOST": f"{profiles_service}.{self.model.name}",
+                        "REGISTRATION_FLOW": self.model.config["registration-flow"],
+                        "DASHBOARD_LINKS_CONFIGMAP": self.model.config[
+                            "dashboard-configmap"
+                        ],
```

Comment:
I know this is what was here before in the pod_spec charm, but dashboard-configmap feels like an odd thing to have as a config option.  Especially once you switch to having a dynamically generated sidebar

Reply:
I mean for me having the configuration option there makes sense if we dont remove existing configmap .... if we always erase than it make less of a sense .... the configmap change is automatically propagated ... you will see in #38 after I close this PR